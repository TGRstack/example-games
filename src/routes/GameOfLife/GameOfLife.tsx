// tslint:disable max-line-length
import Gx from '@tgrx/gx'
import * as React from 'react'
import Helmet from 'react-helmet'
import GameOfLifeBoard from '../../components/game-of-life/LifeBoard'
// import * as S from './GameOfLife.css'

function SimpleWizard() {
  return (
    <div>
      <Helmet>
        <title>Game of Life</title>
      </Helmet>

      <Gx col={12}>
        <GameOfLifeBoard />
      </Gx>
    </div>
  )
}

export default SimpleWizard
