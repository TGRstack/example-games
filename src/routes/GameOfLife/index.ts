import GameOfLife from './GameOfLife'

export default GameOfLife
