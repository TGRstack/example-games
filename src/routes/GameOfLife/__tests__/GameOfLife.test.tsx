/* eslint-disable import/no-extraneous-dependencies */

import { shallow } from 'enzyme'
import * as React from 'react'
import GameOfLife from '../GameOfLife'

describe('<GameOfLife />', () => {
  test('renders', () => {
    const wrapper = shallow(<GameOfLife />)
    expect(wrapper).toMatchSnapshot()
  })
})
