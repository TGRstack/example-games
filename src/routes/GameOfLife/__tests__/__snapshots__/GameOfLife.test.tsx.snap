// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameOfLife /> renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SimpleWizard />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <HelmetWrapper
          defer={true}
          encodeSpecialCharacters={true}
        >
          <title>
            Game of Life
          </title>
        </HelmetWrapper>,
        <Gx
          col={12}
        >
          <LifeBoard
            label="text"
          />
        </Gx>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <title>
            Game of Life
          </title>,
          "defer": true,
          "encodeSpecialCharacters": true,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Game of Life",
          },
          "ref": null,
          "rendered": "Game of Life",
          "type": "title",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <LifeBoard
            label="text"
          />,
          "col": 12,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "label": "text",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <HelmetWrapper
            defer={true}
            encodeSpecialCharacters={true}
          >
            <title>
              Game of Life
            </title>
          </HelmetWrapper>,
          <Gx
            col={12}
          >
            <LifeBoard
              label="text"
            />
          </Gx>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <title>
              Game of Life
            </title>,
            "defer": true,
            "encodeSpecialCharacters": true,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Game of Life",
            },
            "ref": null,
            "rendered": "Game of Life",
            "type": "title",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <LifeBoard
              label="text"
            />,
            "col": 12,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "label": "text",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
